src_MFsurfchem/MFsurfchem_functions.cpp

```diff
@@ -48,13 +52,11 @@ void init_surfcov(MultiFab& surfcov, const amrex::Real* dx)
         Dim3 hi = ubound(bx);
         const Array4<Real> & surfcov_arr = surfcov.array(mfi);
 
-        amrex::Real Ntot = surf_site_num_dens*dx[0]*dx[1];  // total number of reactive sites
-
-        amrex::ParallelForRNG(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k, RandomEngine const& engine) noexcept
+        amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
         {
             if (k==0) {
                 for (int m=0;m<n_ads_spec;m++) {
-		  surfcov_arr(i,j,k,m) = RandomPoisson(Ntot*surfcov0/n_ads_spec,engine)/Ntot;
+                    surfcov_arr(i,j,k,m) = surfcov0[m];
```

Comment:
was this supposed to be
RandomPoisson(Ntot*surfcov0[m]/n_ads_spec,engine)/Ntot;

Reply:
Not at the moment, no. For now, we are simply assigning the initial value of `surfcov0` to the`surfcov_arr` to initialize it.  That is why Ntot is not defined within that function, and the loop is a ParallerFor instead of a ParallelForRNG.